### DISCUSSION TITLE
Recording network requests within intercept

### DISCUSSION THREAD

**Discussion by jfangrad** [cypress-io/cypress#9331] [Questions and Help]
I am working on a project where we need to record all the network requests made during a test suite. With the introduction of v6 and intercept it seemed like a great solution.

However it seems that I can't write files from within an intercept. For example, I am doing the following:
`cypress/support/index.ts`
```js
before(() => {
 ...
  const routeRegex = new RegExp(`${RequestConfig.baseURL}/api`)
  cy.intercept({ url: routeRegex }, e2eFixtureHandler);
});

function e2eFixtureHandler(req: CyHttpMessages.IncomingHttpRequest) {
  routeCounter(req);
  if (RECORD_MODE) {
    recordFixture(req);
  } else {
    req.reply();
  }
}

function recordFixture(
  req: CyHttpMessages.IncomingHttpRequest,
) {
  req.reply((res) => {
    try {
      const { pathToFixture, fixtureFolder } = fixtureInfoFromReq(req);
      const fixture: Fixture = {
        status: res.statusCode,
        data: res.body,
        responseHeaders: res.headers,
      };
      cy.writeFile(path.join('../..', pathToFixture), fixture);
    } catch (err) {
      console.error(err);
    }
  });
}
```

When running the tests I always end up getting these errors though. Is there something I am doing wrong or is it not possible to write files out from within an intercept?
<img width="820" alt="Screen Shot 2020-11-25 at 10 46 27 AM" src="https://user-images.githubusercontent.com/11445215/100250272-7b9d8a80-2f0b-11eb-8f78-b4409dc0624d.png">

**Comment by giuseppeg** [comment]
The answer is here https://www.youtube.com/watch?v=3uQpeyBlw3w
